Warnings now appearing about unused-private-field

Builds now result in this:

```
warning: In file included from /Users/adetaylor/autocxx-demo-4/target/debug/build/autocxx-demo-4-a14d6da754909d48/out/autocxx-build-dir/cxx/gen0.cxx:2:
warning: /Users/adetaylor/autocxx-demo-4/target/debug/build/autocxx-demo-4-a14d6da754909d48/out/autocxx-build-dir/include/autocxxgen.h:9:11: warning: private field 'ptr' is not used [-Wunused-private-field]
warning:     char* ptr;
warning:           ^
warning: /Users/adetaylor/autocxx-demo-4/target/debug/build/autocxx-demo-4-a14d6da754909d48/out/autocxx-build-dir/include/autocxxgen.h:25:11: warning: private field 'ptr' is not used [-Wunused-private-field]
warning:     char* ptr;
warning:           ^
warning: /Users/adetaylor/autocxx-demo-4/target/debug/build/autocxx-demo-4-a14d6da754909d48/out/autocxx-build-dir/include/autocxxgen.h:49:11: warning: private field 'ptr' is not used [-Wunused-private-field]
warning:     char* ptr;
warning:           ^
warning: 3 warnings generated.
warning: In file included from /Users/adetaylor/autocxx-demo-4/target/debug/build/autocxx-demo-4-a14d6da754909d48/out/autocxx-build-dir/cxx/gen1.cxx:1:
warning: /Users/adetaylor/autocxx-demo-4/target/debug/build/autocxx-demo-4-a14d6da754909d48/out/autocxx-build-dir/include/autocxxgen.h:9:11: warning: private field 'ptr' is not used [-Wunused-private-field]
warning:     char* ptr;
warning:           ^
warning: /Users/adetaylor/autocxx-demo-4/target/debug/build/autocxx-demo-4-a14d6da754909d48/out/autocxx-build-dir/include/autocxxgen.h:25:11: warning: private field 'ptr' is not used [-Wunused-private-field]
warning:     char* ptr;
warning:           ^
warning: /Users/adetaylor/autocxx-demo-4/target/debug/build/autocxx-demo-4-a14d6da754909d48/out/autocxx-build-dir/include/autocxxgen.h:49:11: warning: private field 'ptr' is not used [-Wunused-private-field]
warning:     char* ptr;
warning:           ^
warning: 3 warnings generated.


```

This is a regression and I know what I did wrong. Just need to fix it...

--- a/engine/src/lib.rs
+++ b/engine/src/lib.rs
@@ -267,15 +267,13 @@ impl IncludeCppEngine {
     }
 
     fn build_header(&self) -> String {
-        let full_header = join(
+        join(
             self.config.inclusions.iter().map(|incl| match incl {
                 CppInclusion::Define(symbol) => format!("#define {}\n", symbol),
                 CppInclusion::Header(path) => format!("#include \"{}\"\n", path),
             }),
             "",
-        );
-
-        format!("{}\n\n{}", KNOWN_TYPES.get_prelude(), full_header,)
+        )
     }
 
     fn make_bindgen_builder(
@@ -382,7 +380,8 @@ impl IncludeCppEngine {
         }
         let header_contents = self.build_header();
         self.dump_header_if_so_configured(&header_contents, &inc_dirs, &definitions);
-        builder = builder.header_contents("example.hpp", &header_contents);
+        let header_and_prelude = format!("{}\n\n{}", KNOWN_TYPES.get_prelude(), header_contents);
+        builder = builder.header_contents("example.hpp", &header_and_prelude);
 
         let bindings = builder.generate().map_err(Error::Bindgen)?;
         let bindings = self.parse_bindings(bindings)?;
@@ -395,7 +394,7 @@ impl IncludeCppEngine {
                 bindings,
                 self.config.exclude_utilities,
                 self.config.unsafe_policy.clone(),
-                self.build_header(),
+                header_contents,
             )
             .map_err(Error::Conversion)?;
         let mut items = conversion.rs;

